enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,242 @@
+  - "@williamcaban"
+approvers:
+  - "@soltysh"
+  - "@dusk125"
+  - "@hasbro17"
+  - "@tjungblu"
+  - "@williamcaban"
+api-approvers:
+  - "@soltysh"
+creation-date: 2024-06-17
+last-updated: 2024-06-17
+tracking-link:
+  - https://issues.redhat.com/browse/ETCD-609
+see-also:
+  - "https://issues.redhat.com/browse/OCPSTRAT-1411"
+  - "https://issues.redhat.com/browse/OCPSTRAT-529"
+---
+
+
+# Automated Backups of etcd with No Config
+
+## Summary
+
+Enable the automated backups of etcd from day 1 without provided config from the user.
+
+## Motivation
+The [current automated backup of etcd](https://docs.openshift.com/container-platform/4.15/backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.html#creating-automated-etcd-backups_backup-etcd) of an OpenShift
+cluster relies on user provided configurations. 
+
+To improve the customers experience, this work proposes taking etcd backup using default config from day one, without additional configuration from the user.
+
+
+### Goals
+- Backups should be taken without configuration after cluster installation from day 1.
+- Backups are saved to a default PersistentVolume, that could be overridden by user.
+- Backups are taken according to a default schedule, that could be overridden by user.
+- Backups are taken according to a default retention policy, that could be overridden by user.
+
+
+### Non-Goals
+- Save cluster backups to cloud storage e.g S3.
+  - This could be a future enhancement or extension to the API.
+- Automate cluster restoration.
+- Provide automated backups for non-self hosted architectures like Hypershift.
+
+### User Stories
+- As a cluster administrator I want cluster backup to be taken without configuration.
+- As a cluster administrator I want to schedule recurring cluster backups so that I have a recent cluster state to recover from in the event of quorum loss (i.e. losing a majority of control-plane nodes).
+- As a cluster administrator I want to have failure to take cluster backups for more than a configurable period to be reported to me via critical alerts.
+
+
+## Proposal
+- Provide a default etcd backup configuration as a `EtcdBackup` CR.
+- Provide a default reliable storage mechanism to save the backup data on.
+- Provide a separate feature gate for the automated backup with default config (under discussion).
+
+
+### Workflow Description
+- The user will enable the AutomatedBackupNoConfig feature gate (under discussion).
+- A `EtcdBackup` CR is being installed on the cluster.
+- In a default config scenario, the use should not add any config.
+- A PVC and PV are being created to save the backup data reliably.
+
+
+### API Extensions
+- No API changes are required. 
+- See https://github.com/openshift/enhancements/blob/master/enhancements/etcd/automated-backups.md#api-extensions
+
+### Topology Considerations
+TBD
+#### Hypershift / Hosted Control Planes
+TBD
+#### Standalone Clusters
+TBD
+#### Single-node Deployments or MicroShift
+TBD
```

Comment:
I believe the strategy will be different for different topology, so I'd expect an outline how each topology will be handled wrt defaults.

Reply:
So we are planning to use **SideCar Container** within each etcd pod manifest, and the backup will be save to a `hostPath` volume. 

iiuc in this approach the strategy will be the same regardless of the topology, or ?

Reply:
Yes, it should, just make sure to call it out explicitly.

Reply:
If you're taking the exact same approach across all supported topologies, I'd at minimum add here links to the design details, pointing out that the solution is orthogonal to topology of the cluster.

Reply:
Probably not applicable to Microshift - it doesn't use CEO so it can't schedule backups iiuc. Lets mention that in the enhancement.

SNO probably won't be significantly different from HA cluster

Reply:
Yes it is the same approach, that why we took the sidecar approach

Reply:
This comment appears to be unresolved.